packages/client/src/client.ts

```diff
@@ -0,0 +1,98 @@
+import { assertProviderManifest, assertProviderRecord } from './assert.js';
+import { PenumbraProviderNotConnectedError } from './error.js';
+import { PenumbraStateEventDetail, PenumbraStateEvent, isPenumbraStateEvent } from './event.js';
+import { availableOrigin } from './get.js';
+import { PenumbraProvider } from './provider.js';
+import { PenumbraState } from './state.js';
+
+export interface IPenumbraClient {
+  /**
+   * Asks users to approve the connection to a specific browser manifest URL.
+   * If `manifest` argument is not provided, tries to connect to the first injected provider.
```

Comment:
suggestion (idea): 

```suggestion
   * If `manifest` argument is not provided, tries to connect to the previously-connected or the first injected provider.
```

from the implementation point, it's going to look over the `isConnected()` of all injected providers and connect to the first one with `true`